src/og/entity/GeoObjectHandler.js

```diff
@@ -0,0 +1,1106 @@
+    for (let i = currIndex; i < currIndex + length; i++) {
+        arr[i] = params[i % itemSize];
+    }
+    return arr;
+};
+
+class GeoObjectHandler {
+    constructor(entityCollection) {
+        /**
+         * Picking rendering option.
+         * @public
+         * @type {boolean}
+         */
+        this.pickingEnabled = true;
+
+        this._entityCollection = entityCollection;
+        this._renderer = null;
+        this._planet = null;
+
+        this._geoObjects = [];
+        this._pitchRollArr = [[]];
+        this._sizeArr = [[]];
+        this._vertexArr = [[]];
+        this._positionHighArr = [[]];
+        this._positionLowArr = [[]];
+        this._directionArr = [[]];
+        this._rgbaArr = [[]];
+        this._normalsArr = [[]];
+        this._indicesArr = [[]];
+        this._pickingColorArr = [[]];
+
+        this._pitchRollBuffer = [];
+        this._sizeBuffer = [];
+        this._vertexBuffer = [];
+        this._positionHighBuffer = [];
+        this._positionLowBuffer = [];
+        this._directionBuffer = [];
+        this._rgbaBuffer = [];
+        this._normalsBuffer = [];
+        this._indicesBuffer = [];
+        this._pickingColorBuffer = [];
+
+        this.__staticId = GeoObjectHandler._staticCounter++;
+        this._buffersUpdateCallbacks = [];
+        this._buffersUpdateCallbacks[PICKINGCOLOR_BUFFER] = this.createPickingColorBuffer;
+        this._buffersUpdateCallbacks[POSITION_BUFFER] = this.createPositionBuffer;
+        this._buffersUpdateCallbacks[DIRECTION_BUFFER] = this.createDirectionBuffer;
+        this._buffersUpdateCallbacks[NORMALS_BUFFER] = this.createNormalsBuffer;
+        this._buffersUpdateCallbacks[RGBA_BUFFER] = this.createRgbaBuffer;
+        this._buffersUpdateCallbacks[INDECIES_BUFFER] = this.createIndicesBuffer;
+        this._buffersUpdateCallbacks[VERTEX_BUFFER] = this.createVertexBuffer;
+        this._buffersUpdateCallbacks[SIZE_BUFFER] = this.createSizeBuffer;
+        this._buffersUpdateCallbacks[PITCH_ROLL_BUFFER] = this.createPitchRollBuffer;
+
+        this._changedBuffers = new Array(this._buffersUpdateCallbacks.length);
+
+        this._instancedTags = new Map();
+        this._instancedTags.set("none", {
+            index: 0
+        });
+    }
+
+    getObjectByIndex(index) {
+        return this._geoObjects[index];
+    }
+
+    //Create buffers
+    createVertexBuffer() {
+        var h = this._renderer.handler;
+        for (let i = 0; i < this._vertexArr.length; i++) {
+            this._vertexBuffer && h.gl.deleteBuffer(this._vertexBuffer[i]);
+            this._vertexArr[i] = makeArrayTyped(this._vertexArr[i]);
+            this._vertexBuffer[i] = h.createArrayBuffer(
+                this._vertexArr[i],
+                3,
+                this._vertexArr[i].length / 3
+            );
+        }
+    }
+
+    createPitchRollBuffer() {
+        var h = this._renderer.handler;
+        for (let i = 0; i < this._pitchRollArr.length; i++) {
```

Comment:
We do not need to recreate all buffers when we change only one tag group.

Reply:
I'm plane use first item of arrays for uninstanced objects. But its difficult and it's was before moment, when we talk about this feature